mlflow/server/js/src/experiment-tracking/components/MetricsPlotPanel.tsx

```diff
@@ -462,145 +448,23 @@ export class MetricsPlotPanel extends React.Component<MetricsPlotPanelProps, Met
     return state.layout && state.layout.yaxis && state.layout.yaxis.type === 'log' ? 'log' : 'linear';
   }
 
-  /**
-   * Handle changes to metric plot layout (x & y axis ranges), e.g. specifically if the user
-   * zooms in or out on the plot.
-   *
-   * @param newLayout: Object containing the new Plot layout. See
-   * https://plot.ly/javascript/plotlyjs-events/#update-data for details on the object's fields
-   * and schema.
-   */
-  handleLayoutChange = (newLayout: any) => {
-    this.displayPopover = false;
-    const state = this.getUrlState();
-    // Unfortunately, we need to parse out the x & y axis range changes from the onLayout event...
-    // see https://plot.ly/javascript/plotlyjs-events/#update-data
-    const {
-      'xaxis.range[0]': newXRange0,
-      'xaxis.range[1]': newXRange1,
-      'yaxis.range[0]': newYRange0,
-      'yaxis.range[1]': newYRange1,
-      'xaxis.autorange': xAxisAutorange,
-      'yaxis.autorange': yAxisAutorange,
-      'yaxis.showspikes': yAxisShowSpikes,
-      'xaxis.showspikes': xAxisShowSpikes,
-      ...restFields
-    } = newLayout;
-
-    let mergedLayout = {
-      ...state.layout,
-      ...restFields,
-    };
-    let lastLinearYAxisRange = [...(state as any).lastLinearYAxisRange];
-
-    // Set fields for x axis
-    const newXAxis = mergedLayout.xaxis || {};
-    if (newXRange0 !== undefined && newXRange1 !== undefined) {
-      newXAxis.range = [newXRange0, newXRange1];
-      newXAxis.autorange = false;
-    }
-    if (xAxisShowSpikes) {
-      newXAxis.showspikes = true;
-    }
-    if (xAxisAutorange) {
-      newXAxis.autorange = true;
-    }
-    // Set fields for y axis
-    const newYAxis = mergedLayout.yaxis || {};
-    if (newYRange0 !== undefined && newYRange1 !== undefined) {
-      newYAxis.range = [newYRange0, newYRange1];
-      newYAxis.autorange = false;
-    }
-    if (yAxisShowSpikes) {
-      newYAxis.showspikes = true;
-    }
-    if (yAxisAutorange) {
-      lastLinearYAxisRange = [];
-      const axisType = state.layout && state.layout.yaxis && state.layout.yaxis.type === 'log' ? 'log' : 'linear';
-      newYAxis.autorange = true;
-      newYAxis.type = axisType;
+  handleDownloadCsv = async () => {
+    const { loading } = this.state;
+    if (!loading) {
+      const state = this.getUrlState();
+      const selectedMetricKeys = state.selectedMetricKeys || [];
+      this.loadMetricHistoryPromise = this.loadMetricHistory(this.props.runUuids, selectedMetricKeys);
```

Comment:
nit: isn't `loadMetricHistory` setting this instance field anyway?
shouldn't we just `await this.loadMetricHistory(this.props.runUuids, selectedMetricKeys);`?